File: C++/Easy/WrongAnswer/check-whether-two-strings-are-almost-equivalent_7.cpp
class Solution {
public:
    bool checkAlmostEquivalent(string word1, string word2) {
        
        int n = word1.size();
        if (n != word2.size()) {
            return false;
        }
        vector<int> c1(26, 0), c2(26, 0);
        for (int i = 0; i < n; ++i) {
            c1[word1[i] - 'a']++;
            c2[word2[i] - 'a']++;
        }
        int diff = 0;
        for (int i = 0; i < 26; ++i) {
            diff += abs(c1[i] - c2[i]);
        }
        return diff <= 3;
    }
};

File: Rust/Easy/WrongAnswer/check-whether-two-strings-are-almost-equivalent_7.rs
impl Solution {
    pub fn check_almost_equivalent(word1: String, word2: String) -> bool {
        
        let mut word1 = word1.chars().collect::<Vec<char>>();
        let mut word2 = word2.chars().collect::<Vec<char>>();
        word1.sort();
        word2.sort();
        let mut i = 0;
        let mut j = 0;
        let mut diff = 0;
        while i < word1.len() && j < word2.len() {
            if word1[i] == word2[j] {
                i += 1;
                j += 1;
            } else if word1[i] != word2[j] {
                diff += 1;
                if diff > 3 {
                    return false;
                }
                if i + 1 < word1.len() && word1[i] == word1[i + 1] {
                    i += 1;
                } else if j + 1 < word2.len() && word2[j] == word2[j + 1] {
                    j += 1;
                } else {
                    i += 1;
                    j += 1;
                }
            }
        }
        if i < word1.len() || j < word2.len() {
            diff += 1;
        }
        diff <= 3
    }
}